controllers/controlplane_controller.go

```diff
@@ -67,12 +67,39 @@ func (r *ControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, nil // no need to requeue, the update will trigger.
 	}
 
+	debug(log, "validating ControlPlane's DataPlane configuration", controlplane)
+	if err = r.ensureDataPlaneConfiguration(ctx, controlplane, nil); err != nil {
+		if errors.IsConflict(err) {
+			debug(log, "conflict found when updating ControlPlane resource, retrying", controlplane)
```

Comment:
We've got three different verbs used here which could add to some confusion for future readers:

1. validate
2. ensure
3. update

we should try to update the language to be consistent from line to line, perhaps:

```suggestion
	debug(log, "ensuring that the ControlPlane DataPlane configuration is up to date", controlplane)
	if err = r.ensureDataPlaneConfiguration(ctx, controlplane, nil); err != nil {
		if errors.IsConflict(err) {
			debug(log, "conflict found when trying to ensure ControlPlane configuration was up to date, retrying", controlplane)
```

I'm not positive that the language is quite what we want either, but just to illustrate the point.

Reply:
I probably wrote those lines :joy: I do think it's reasonable to challenge everything, but if you don't want to that's fine. Consider this comment resolved at your own discretion as it's really a nitpick anyhow.

Reply:
For me, if we can make it better it great. As long as we don't break anything :D